browser/profiles/brave_renderer_updater.cc

```diff
@@ -107,10 +122,12 @@ void BraveRendererUpdater::UpdateRenderer(
            brave_wallet::mojom::DefaultWallet::BraveWalletPreferExtension ||
        default_ethereum_wallet ==
            brave_wallet::mojom::DefaultWallet::BraveWallet) &&
-      is_wallet_allowed_for_context_ && brave_wallet::IsDappsSupportEnabled();
+      is_wallet_allowed_for_context_ && brave_wallet::IsDappsSupportEnabled() &&
+      !should_ignore_brave_wallet;
```

Comment:
We shouldn't apply this when default wallet == `brave_wallet::mojom::DefaultWallet::BraveWallet`